apel/parsers/arc.py

```diff
@@ -0,0 +1,72 @@
+from apel.common.datetime_utils import parse_timestamp
+from apel.db.records.job import JobRecord
+from apel.parsers import Parser
+
+
+class ARCParser(Parser):
+    """
+    Parser for ARC accounting records.
+
+    This parser differs from most of the others as ARC saves each accounting
+    record in a separate file.
+    """
+    def parse_arc_file(self, arcfile):
+        """
+        Parse an ARC accounting file.
+
+        Takes an open ARC accounting file object and returns a JobRecord and the
+        number of lines in it.
+
+        The ARC to APEL mapping used can be found in the JURA souce code here:
+        http://svn.nordugrid.org/trac/nordugrid/browser/arc1/trunk/src/services/a-rex/jura/arexlog2ur.txt
+        """
+        arcjob = {}
+        lines = 0
+        for line in arcfile:
+            # Remove any leading or trailing whitespace including newlines.
+            line = line.strip()
+            lines += 1
+            key, value = line.split('=', 1)
+            arcjob[key] = value
+
+        if lines == 0 or 'status' not in arcjob or arcjob['status'] != "completed":
+            # Empty file, incomplete job, or failed job (respectively)
+            return None, lines
+
+        arcjob['accounting_options'] = self.parse_accounting_options(arcjob['accounting_options'])
+
+        apeljob = {}
```

Comment:
This could be rewritten as:
``apeljob = {
   'Site': arcjob['accounting_options']['gocdb_name'],
   'SubmitHost': arcjob['headnode']
``
and so on. There is a lot of repetitions of word 'apeljob'.